frontend/src/components/PushPopover.vue

```diff
@@ -0,0 +1,242 @@
+// - Add a mixpanel event when a CTA is clicked (with the CTA type?)
+
+import {computed, ref} from 'vue';
+import {usePushStore} from "@/stores/push";
+import {useMainStore} from "@/stores/main";
+import {useCatalogStore} from "@/stores/catalog";
+import {storeToRefs} from "pinia";
+import {useProjectStore} from "@/stores/project";
+import {useDebuggingSessionsStore} from "@/stores/debugging-sessions";
+import {$vfm} from "vue-final-modal";
+import AddTestToSuite from "@/views/main/project/modals/AddTestToSuite.vue";
+import {chain} from "lodash";
+import {TYPE} from "vue-toastification";
+import {RowFilterType} from "@/generated-sources";
+
+const pushStore = usePushStore();
+const mainStore = useMainStore();
+const {slicingFunctionsByUuid, testFunctionsByUuid} = storeToRefs(useCatalogStore())
+const catalogStore = useCatalogStore();
+const projectStore = useProjectStore();
+const debuggingStore = useDebuggingSessionsStore();
+
+//:modelFeatures="modelFeatures"
+//:inputData="inputData"
+const props = defineProps({
+  type: String,
+  column: String
+});
+
+const opened = ref<boolean>(false);
+const loading = ref<string>("");
+const value = computed(() => {
+  return pushStore.current;
+});
+
+const show = computed(() => {
+  switch (props.type) {
+    case "contribution":
+      return value.value?.contribution.key == props.column;
+    case "perturbation":
+      return value.value?.perturbation.key == props.column;
+    case "overconfidence":
+      return value.value?.overconfidence.pushTitle && value.value?.overconfidence.pushTitle != "";
+    case "borderline":
+      return value.value?.borderline.pushTitle && value.value?.borderline.pushTitle != "";
+    default:
+      return false;
+  }
+})
+const push = computed(() => {
+  switch (props.type) {
+    case "contribution":
+      return value.value?.contribution;
+    case "perturbation":
+      return value.value?.perturbation;
+    case "overconfidence":
+      return value.value?.overconfidence;
+    case "borderline":
+      return value.value?.borderline;
+    default:
+      return false;
+  }
+})
+
+const icon = computed(() => {
+  switch (props.type) {
+    case "contribution":
+      return "mdi-chart-bar";
+    case "perturbation":
+      return "mdi-waveform";
+    case "overconfidence":
+      return "mdi-chevron-triple-up";
+    case "borderline":
+      return "mdi-approximately-equal";
+    default:
+      return "mdi-bug";
+  }
+});
+
+async function applyCta(kind: string) {
+  loading.value = kind;
+  // modelId: string, datasetId: string, rowNb: number, pushKind: string, ctaKind: string
+  // @ts-ignore
+  let uuid = await pushStore.applyPush(push.value!.kind, kind);
+
+
+  switch (kind) {
+    case "CreateSlice":
+    case "SaveExample":
+      mainStore.addSimpleNotification("Successfully saved");
+      break;
+    case "CreateSliceOpenDebugger":
+      await catalogStore.loadCatalog(projectStore.currentProjectId ?? 0);
```

Comment:
I think it'll be clearer to break down each cases into method and just calling them.

Furthermore I also think `kind` should be an enum instead of a string